src/components/stimuli/Stimuli.tsx

```diff
@@ -29,15 +44,80 @@ const Stimuli: React.FC = () => {
       }
     }
 
-    fetchStimuliUrls();
-  }, []);
+    if (!isUploading) fetchStimuliUrls();
+  }, [isUploading]);
+
+  const onFileSelect = (e: any) => {
+    const file: File = e.target.files[0];
+    const formData = new FormData();
+    formData.append('file', file);
+    setUploading(true);
+    fetch(`${process.env.REACT_APP_BACKEND_ADDRESS}/image`, {
+      method: 'POST',
+      body: formData,
+    })
+      .then(res => res.json())
+      .then(() => {
+        setUploading(false);
+        setShowToast(true);
+      });
+  };
```

Comment:
We should probably use async rather than promise chaining for consistency with other async calls. See how `fetchStimuliURLs` looks for an example.

We should include a try-catch block for the case where an upload fails for some reason.